client/src/NeedsAndOffers.tsx

```diff
@@ -45,19 +46,23 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 
 type Props = {
-  headerText: string,
-  cards: Asset[],
+    cards: Asset[],
+    headerContentRight?: JSX.Element,
```

Comment:
yes. the `?` tells the component that the prop is "optional". so sometimes it will have a value passed in as that prop and sometimes not. and so that's why i have to check to make sure there is a value before i render it with that `{headerContentRight ?? null}`

the `??` is the "null coalesce" operator, so if what is on the left evaluates to null then we fallback to the right side. so basically saying "render this headerContentRight or nothing"

p.s. use backticks ` to make a snippet in here or triple backticks for a multiline snippet